### PR TITLE
JP tagger recalculation

### PR DESCRIPTION
**Pull request by colizz**
Adding the calculation of jet probability (JP) and jet B probability (JBP).
This is the exact implementation in cmssw, but now written in coffea code.

For each AK4 jet, we take the matched tracks from the PFCands collection. 
The IPsig templates as "JPCalibration" file are required to derive the track probability. These track probabilities are then used to calculate jet probability.

### CODE REVIEW

File: src/BTVNanoCommissioning/workflows/BTA_producer.py [hunk 1]
```diff
@@ -469,6 +518,250 @@ def process_shift(self, events, shift_name):
...
+        ]  # select up to 4 tracks
+
+        # for positive JBP tagger: probabilities include both pos+neg IP tracks; for negative JBP: use neg IP tracks only
+        Jet["Bprob"] = (
+            -np.log(jpc.calc_jet_proba(trk_Bproba_pos)) / 4.0
+            - np.log(jpc.calc_jet_proba(trk_proba_all)) / 4.0
+        )
+        Jet["BprobN"] = (
+            -np.log(jpc.calc_jet_proba(trk_Bproba_neg)) / 4.0
+            - np.log(jpc.calc_jet_proba(trk_proba_neg)) / 4.0
```
Comment (Ming-Yan) [inline src/BTVNanoCommissioning/workflows/BTA_producer.py:759]: `jpc.calc_jet_proba(trk_proba_neg)` appears twice in the code, I am wonder whether we can simplify this by `Jet["ProbaN"]*np.log(10)` to transfer to `- np.log(jpc.calc_jet_proba(trk_proba_neg)) / 4.0`

**Review by Ming-Yan — COMMENTED** [review]
Hi @colizz  thank you for the PR on the JP recalculation! 
Sorry I take a bit longer trying to figure out the possibility to get rid of the array builder if possible.

I only have some minor comments regarding to the restructure of the files and directory.
Please let me know if you have any suggestions! (I could also made these changes and merge it)
Thank you !

File: src/BTVNanoCommissioning/workflows/BTA_producer.py (hunk 1 above)
Comment (Ming-Yan) [reply src/BTVNanoCommissioning/workflows/BTA_producer.py:759]: Thanks for the clarification I misunderstood the differences between `track_proba_neg & track_jet_proba_neg`

**Comment by colizz** [comment]
Hi, I made some changes as recommended. I also managed to use the columnar operations (removing the array builder) in the JP calculation.

**Pull request merged**